tests/client/test_auth.py

```diff
@@ -89,80 +89,86 @@ def challenge_send(self, request: AsyncRequest) -> AsyncResponse:
         headers = [
             ("www-authenticate", 'Digest realm="httpx@example.org", ' + challenge_str)
         ]
-        return AsyncResponse(401, headers=headers, content=b"", request=request)
+        return Response(401, headers=headers, content=b"", request=request)
 
 
-def test_basic_auth():
+@pytest.mark.asyncio
+async def test_basic_auth():
     url = "https://example.org/"
     auth = ("tomchristie", "password123")
 
-    with Client(dispatch=MockDispatch()) as client:
-        response = client.get(url, auth=auth)
+    client = Client(dispatch=MockDispatch())
+    response = await client.get(url, auth=auth)
```

Comment:
Thoughts on keeping `async with` usage here and elsewhere, out of consistency?

Reply:
Really not needed. We’re not closing a connection pool, so I think we should keep the style simple.